Sources/TuistDoc/SwiftDocServer.swift

```diff
@@ -0,0 +1,130 @@
+
+    /// Utility to open files
+    private let opener: Opening
+
+    /// HTTPServer from Swifter
+    private var server: HttpServer?
+
+    /// Semaphore to block the execution
+    private var semaphore: DispatchSemaphore?
+
+    public let baseURL: String = "http://localhost"
+
+    public init(fileHandling: FileHandling = FileHandler(),
+                opener: Opening = Opener())
+    {
+        self.fileHandling = fileHandling
+        self.opener = opener
+    }
+
+    public func serve(path: AbsolutePath, port: UInt16) throws {
+        SwiftDocServer.temporaryDirectory = path
+
+        func okReponse(fileAt path: AbsolutePath) throws -> HttpResponse {
+            guard let file = try? path.pathString.openForReading() else {
+                return .notFound
+            }
+            return .raw(200, "OK", [:]) { writer in
+                try? writer.write(file)
+                file.close()
+            }
+        }
+
+        func handleRequest(_ request: HttpRequest) -> HttpResponse {
+            guard let (_, value) = request.params.first else {
+                return .notFound
+            }
+
+            let filePath = path.appending(component: value)
+            guard fileHandling.exists(filePath) else { return .notFound }
+
+            do {
+                if try filePath.pathString.directory() {
+                    // this is how swift-doc generates it
+                    let indexPath = filePath
+                        .appending(component: SwiftDocServer.index)
+                    return try okReponse(fileAt: indexPath)
+                } else {
+                    return try okReponse(fileAt: filePath)
+                }
+            } catch {
+                return .internalServerError
+            }
+        }
+
+        server = HttpServer()
+
+        server?["/:param"] = handleRequest
+
+        Signals.trap(signals: [.int, .abrt]) { _ in
+            // swiftlint:disable:next force_try
+            logger.pretty("Deleting temporary folder.")
+            try? SwiftDocServer.temporaryDirectory.map(FileHandler.shared.delete)
+            exit(0)
+        }
+
+        guard let serverURL = URL(string: baseURL + ":\(port)") else { throw Error.unableToStartServer(at: port) }
+
+        semaphore = DispatchSemaphore(value: 0)
+        do {
+            logger.pretty("Starting server at \(.bold(.raw(serverURL.absoluteString))).")
+            try server?.start(port, forceIPv4: true)
+
+            let urlPath = serverURL.appendingPathComponent(SwiftDocServer.index)
+            logger.pretty("Opening the documentation. Press \(.keystroke("CTRL + C")) once you are done.")
+            try opener.open(url: urlPath)
+
+            semaphore?.wait()
+        } catch {
+            logger.error("Server start error: \(error)")
+            semaphore?.signal()
+            throw Error.unableToStartServer(at: port)
+        }
+    }
+}
+
+// MARK: - Error
+
+extension SwiftDocServer {
```

Comment:
Nitpick: I'd be consistent with how we define errors in the codebases and declare the error at the top avoiding extensions:

```swift
extension SwiftDocServerError: FatalError {
  // Content
}
```